File: android/ValleCOM/app/src/main/java/com/valleapp/valleCOM/adaptadores/AdaptadorTicket.java
package com.valleapp.valleCOM.adaptadores;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import com.valleapp.valleCOM.R;

/**
 * Created by valle on 15/09/14.
 */
public class AdaptadorTicket extends ArrayAdapter<JSONObject> {

    private final Context context;
    private java.util.List<JSONObject> values;

    public AdaptadorTicket(Context context, ArrayList<JSONObject> values) {
        super(context, R.layout.linea_art, (java.util.List<JSONObject>) values);
        this.context = context; this.values = values;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.linea_art, parent, false);
        try {
            TextView can = rowView.findViewById(R.id.lblCan);
            TextView nombre = rowView.findViewById(R.id.lblDescripcion);
            TextView p =  rowView.findViewById(R.id.lblPrecio);
            TextView t =  rowView.findViewById(R.id.lblTotal);
            can.setText(String.format("%s", values.get(position).getString("Can")));
            nombre.setText(String.format("%s",values.get(position).getString("descripcion_t")));
            p.setText(String.format("%.2f €", values.get(position).getDouble("Precio")));
            t.setText(String.format("%.2f €", values.get(position).getDouble("Total")));
        } catch (JSONException e) {
            e.printStackTrace();
        }


        return rowView;
    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorTicket.kt
package com.valleapp.vallecom.adaptadores

import android.annotation.SuppressLint
import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.valleapp.valletpv.R
import org.json.JSONException
import org.json.JSONObject

class AdaptadorTicket(
    private val context: Context,
    private val values: List<JSONObject>
) : RecyclerView.Adapter<AdaptadorTicket.TicketViewHolder>() {

    inner class TicketViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val cantidadTextView: TextView = view.findViewById(R.id.lblCan)
        val descripcionTextView: TextView = view.findViewById(R.id.lblDescripcion)
        val precioTextView: TextView = view.findViewById(R.id.lblPrecio)
        val totalTextView: TextView = view.findViewById(R.id.lblTotal)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TicketViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.linea_art, parent, false)
        return TicketViewHolder(view)
    }

    @SuppressLint("DefaultLocale")
    override fun onBindViewHolder(holder: TicketViewHolder, position: Int) {
        try {
            val item = values[position]
            holder.cantidadTextView.text = item.getString("Can")
            holder.descripcionTextView.text = item.getString("descripcion_t")
            holder.precioTextView.text = String.format("%.2f €", item.getDouble("Precio"))
            holder.totalTextView.text = String.format("%.2f €", item.getDouble("Total"))
        } catch (e: JSONException) {
            Log.e("AdaptadorTicket", "Error al enlazar datos: ${e.message}", e)
        }
    }

    override fun getItemCount(): Int {
        return values.size
    }

    fun toListAdapter(): android.widget.BaseAdapter {
        return object : android.widget.BaseAdapter() {
            override fun getCount(): Int = values.size

            override fun getItem(position: Int): JSONObject = values[position]

            override fun getItemId(position: Int): Long = position.toLong()

            @SuppressLint("DefaultLocale")
            override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
                val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.linea_art, parent, false)
                val cantidadTextView: TextView = view.findViewById(R.id.lblCan)
                val descripcionTextView: TextView = view.findViewById(R.id.lblDescripcion)
                val precioTextView: TextView = view.findViewById(R.id.lblPrecio)
                val totalTextView: TextView = view.findViewById(R.id.lblTotal)

                try {
                    val item = getItem(position)
                    cantidadTextView.text = item.getString("Can")
                    descripcionTextView.text = item.getString("descripcion_t")
                    precioTextView.text = String.format("%.2f €", item.getDouble("Precio"))
                    totalTextView.text = String.format("%.2f €", item.getDouble("Total"))
                } catch (e: JSONException) {
                    Log.e("AdaptadorTicket", "Error al enlazar datos: ${e.message}", e)
                }

                return view
            }
        }
    }
}
